### PR TITLE
Clean up authz authentication flow

### PR DESCRIPTION
**Pull request by nmattia**
This refactors the authentication flow for authz, by making it more consistent with the other (non-authz, i.e. manage) authentication flow and by making the code a bit simpler and more robust. ~~This also fixes an issue where known dapps were not randomized.~~

### CODE REVIEW

File: src/frontend/src/flows/authorize/index.ts [hunk 1]
```diff
@@ -138,14 +138,12 @@ export const authnTemplateAuthorize = ({
  */
 export const authFlowAuthorize = async (
   connection: Connection
-): Promise<void> => {
+): Promise<never> => {
   const i18n = new I18n();
-  const container = document.getElementById("pageContent") as HTMLElement;
+  const dapps = shuffleArray(getDapps());
```
Comment (frederikrothenberger) [inline src/frontend/src/flows/authorize/index.ts:143]: You don't need to shuffle here, right? Only a single logo is needed.

**Review by frederikrothenberger — APPROVED** [review]
LGTM, thanks

File: src/frontend/src/flows/authorize/index.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/flows/authorize/index.ts:143]: ah, that's why it wasn't being shuffled. Thanks!

Comment (nmattia) [reply src/frontend/src/flows/authorize/index.ts]: Clarified (inlined). Thanks!

**Pull request merged**